src/Google.Cloud.Functions.Invoker.Testing/FunctionTestBase.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            Server = server;
+        }
+
+        /// <summary>
+        /// Constructs a new instance using a new server constructed with default settings.
+        /// </summary>
+        protected FunctionTestBase() : this(new FunctionTestServer<TFunction>())
+        {
+        }
+
+        /// <summary>
+        /// Executes a simple HTTP GET request against an arbitrary URI, and returns
+        /// the text of the response after validating that it indicates success. For
+        /// more complex scenarios, use <see cref="ExecuteHttpRequestAsync(HttpRequestMessage, Action{HttpResponseMessage})"/>.
+        /// </summary>
+        public async Task<string> ExecuteHttpGetRequestAsync()
+        {
+            using var client = Server.CreateClient();
+            using var response = await client.GetAsync("uri");
+            response.EnsureSuccessStatusCode();
+            return await response.Content.ReadAsStringAsync();
+        }
+
+        /// <summary>
+        /// Executes the given HTTP request against the server, calling the validators against the response.
+        /// </summary>
+        public async Task ExecuteHttpRequestAsync(HttpRequestMessage request, Action<HttpResponseMessage> validator)
+        {
+            using var client = Server.CreateClient();
+            using var response = await client.SendAsync(request);
+            validator(response);
+        }
+
+        /// <summary>
+        /// Executes the function with the specified CloudEvent, which is converted
+        /// to an HTTP POST request using structured encoding. This method asserts that the
+        /// request completed successfully.
+        /// </summary>
+        public async Task ExecuteCloudEventRequestAsync(CloudEvent cloudEvent)
+        {
+            var bytes = s_eventFormatter.EncodeStructuredEvent(cloudEvent, out var contentType);
+            contentType ??= s_applicationJsonContentType;
+            var mediaContentType = new MediaTypeHeaderValue(contentType.MediaType) { CharSet = contentType.CharSet };
+            var request = new HttpRequestMessage
+            {
+                // Any URI 
+                RequestUri = new Uri("uri", UriKind.Relative),
+                // CloudEvent headers
+                Content = new ByteArrayContent(bytes) { Headers = { ContentType = mediaContentType } },
+                Method = HttpMethod.Post
+            };
+
+            var client = Server.CreateClient();
+            var response = await client.SendAsync(request);
+            response.EnsureSuccessStatusCode();
+        }
+
+        /// <summary>
+        /// Returns a list of the log entries for the category associated with the function.
+        /// </summary>
+        /// <returns>A list of log entries. The returned value may refer to an empty list,
+        /// but is never null.</returns>
+        public List<TestLogEntry> GetFunctionLogEntries() => Server.GetFunctionLogEntries();
```

Comment:
Given that you are adding this one helper here, consider adding another for Server.ClearLogs().